display_list/geometry/dl_region_unittests.cc

```diff
@@ -172,34 +378,54 @@ TEST(DisplayListRegion, TestAgainstSkRegion) {
     std::seed_seq seed{::testing::UnitTest::GetInstance()->random_seed()};
     std::mt19937 rng(seed);
 
-    SkRegion sk_region;
+    SkRegion sk_region1;
+    SkRegion sk_region2;
 
     std::uniform_int_distribution pos(0, 4000);
     std::uniform_int_distribution size(1, settings.max_size);
 
-    std::vector<SkIRect> rects_in;
+    std::vector<SkIRect> rects_in1;
+    std::vector<SkIRect> rects_in2;
 
     for (size_t i = 0; i < settings.iteration_count; ++i) {
       SkIRect rect =
           SkIRect::MakeXYWH(pos(rng), pos(rng), size(rng), size(rng));
-      rects_in.push_back(rect);
-      sk_region.op(rect, SkRegion::kUnion_Op);
+      rects_in1.push_back(rect);
+      sk_region1.op(rect, SkRegion::kUnion_Op);
```

Comment:
SkRegion has a setRects as well, but unfortunately it takes a raw array rather than a vector.

Reply:
It iterates through rects and [adds each one individually](https://github.com/google/skia/blob/9dfea08c013c7a801af7b0e6aae51fc6acba2029/src/core/SkRegion.cpp#L652-L662):

```cpp
bool SkRegion::setRects(const SkIRect rects[], int count) {
    if (0 == count) {
        this->setEmpty();
    } else {
        this->setRect(rects[0]);
        for (int i = 1; i < count; i++) {
            this->op(rects[i], kUnion_Op);
        }
    }
    return !this->isEmpty();
}
```

Weird thing is that there was [optimized version at one point](https://github.com/google/skia/commit/097a3513535ad854c1b049c32c080ec875ab1411) but it was later removed for simple loop when upstreaming android changes.

Reply:
True, I wasn't recommending it as an optimization, but more so that the code was more stylistically symmetric. It's not perfect symmetry because DlRegion provides a 1-step factory and SkRegion would require construction and then setting the rects. I just realized that the raw array arguments aren't an issue because it could be `(rects.data(), rects.size())`

Reply:
I see. I'll change it to `setRects`, it's going to be easier to follow.